CMakeLists.txt

```diff
@@ -0,0 +1,713 @@
+    WORKING_DIRECTORY ${silo_build_include_dir}
+    COMMENT "Creating lite_score.h using mklite"
+    VERBATIM
+    USES_TERMINAL)
+
+##
+# Silo library
+##
+
+# Set up silo library source locations
+set(silo_library_sources
+    ${Silo_SOURCE_DIR}/src/debug/silo_debug.c
+    ${Silo_SOURCE_DIR}/src/filters/f_ale3d.c
+    ${Silo_SOURCE_DIR}/src/filters/f_sample.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdb.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdbdir.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdbio.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdbmm.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdbx.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdconv.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdlow.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdmemb.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdpath.c
+    ${Silo_SOURCE_DIR}/src/pdb/pdrdwr.c
+    ${Silo_SOURCE_DIR}/src/pdb_drv/silo_pdb.c
+    ${Silo_SOURCE_DIR}/src/score/memmove.c
+    ${Silo_SOURCE_DIR}/src/score/scctl.c
+    ${Silo_SOURCE_DIR}/src/score/scctla.c
+    ${Silo_SOURCE_DIR}/src/score/schash.c
+    ${Silo_SOURCE_DIR}/src/score/scstr.c
+    ${Silo_SOURCE_DIR}/src/silo/alloc.c
+    ${Silo_SOURCE_DIR}/src/silo/extface.c
+    ${Silo_SOURCE_DIR}/src/silo/silo.c
+    ${Silo_SOURCE_DIR}/src/silo/silo_ns.c
+    ${Silo_SOURCE_DIR}/src/silo/utils.c
+    ${Silo_SOURCE_DIR}/src/taurus/silo_taurus.c
+    ${Silo_SOURCE_DIR}/src/taurus/taurus.c
+    ${Silo_SOURCE_DIR}/src/unknown/silo_unknown.c)
+
+if(SILO_ENABLE_NETCDF)
+    set(HAVE_NETCDF_DRIVER 1)
+    list(APPEND silo_library_sources 
+        ${Silo_SOURCE_DIR}/src/netcdf/api.c
+        ${Silo_SOURCE_DIR}/src/netcdf/ent.c
+        ${Silo_SOURCE_DIR}/src/netcdf/netcdf.c
+        ${Silo_SOURCE_DIR}/src/netcdf/obj.c
+        ${Silo_SOURCE_DIR}/src/netcdf/silo_netcdf.c
+        ${Silo_SOURCE_DIR}/src/netcdf/table.c)
+endif()
+
+if(SILO_ENABLE_FORTRAN)
+    list(APPEND silo_library_sources ${Silo_SOURCE_DIR}/src/silo/silo_f.c)
+endif()
+
+if(SILO_ENABLE_JSON)
+# is there more to be done to get this to compile?
+#    list(APPEND silo_library_sources ${Silo_SOURCE_DIR}/src/silo/silo_json.c)
+endif()
+
+set(silo_library_include_dirs 
+    ${silo_build_include_dir}
+    ${Silo_SOURCE_DIR}/src/debug
+    ${Silo_SOURCE_DIR}/src/filters
+    ${Silo_SOURCE_DIR}/src/netcdf
+    ${Silo_SOURCE_DIR}/src/pdb
+    ${Silo_SOURCE_DIR}/src/pdb_drv
+    ${Silo_SOURCE_DIR}/src/score
+    ${Silo_SOURCE_DIR}/src/silo
+    ${Silo_SOURCE_DIR}/src/taurus_drv
+    ${Silo_SOURCE_DIR}/src/unknown)
+
+set(SILO_COMPILE_DEFINES SILO_EXPORTS LITE_EXPORTS) 
+
+
+##
+# Add hdf5 support
+##
+if(SILO_ENABLE_HDF5 AND (TARGET hdf5 OR HDF5_FOUND))
+    list(APPEND silo_library_sources 
+        ${Silo_SOURCE_DIR}/src/hdf5_drv/H5FDsilo.c
+        ${Silo_SOURCE_DIR}/src/hdf5_drv/silo_hdf5.c)
+
+    list(APPEND silo_library_include_dirs
+        ${Silo_SOURCE_DIR}/src/hdf5_drv)
+
+    list(APPEND SILO_COMPILE_DEFINES
+        AS_SILO_BUILTIN
+        BIT_STREAM_WORD_TYPE=uint8)
+
+    if(SILO_ENABLE_ZFP)
+        set(HAVE_ZFP 1)
+        list(APPEND silo_library_sources 
```

Comment:
I wonder about having to keep these lists of files consistent with whatever files get added or removed. That comment is specific to just this list of files for ZFP...it goes for all the cases were we enumerate list of source files instead of, for example, the dirs containing them. Honestly, I don't think this is any different from how normal Makefiles work but I did wanna at least ask...

Is there anyway to gather them from a file glob operation on a path (or set of paths)?